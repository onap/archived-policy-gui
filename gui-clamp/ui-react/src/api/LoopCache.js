/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019 AT&T Intellectual Property. All rights
 *                             reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

export default class LoopCache {
	loopJsonCache;

	constructor(loopJson) {
		this.loopJsonCache=loopJson;
	}

	updateMicroServiceProperties(name, newMsProperties) {
			for (var policy in this.loopJsonCache["microServicePolicies"]) {
				if (this.loopJsonCache["microServicePolicies"][policy]["name"] === name) {
					this.loopJsonCache["microServicePolicies"][policy]["configurationsJson"] = newMsProperties;
				}
			}
	}

	updateMicroServicePdpGroup(name, pdpGroup, pdpSubgroup) {
			for (var policy in this.loopJsonCache["microServicePolicies"]) {
				if (this.loopJsonCache["microServicePolicies"][policy]["name"] === name) {
					this.loopJsonCache["microServicePolicies"][policy]["pdpGroup"] = pdpGroup;
					this.loopJsonCache["microServicePolicies"][policy]["pdpSubgroup"] = pdpSubgroup;
				}
			}
	}

	updateGlobalProperties(newGlobalProperties) {
		this.loopJsonCache["globalPropertiesJson"] = newGlobalProperties;
	}

	updateOperationalPolicyProperties(name, newOpProperties) {
		for (var policy in this.loopJsonCache["operationalPolicies"]) {
				if (this.loopJsonCache["operationalPolicies"][policy]["name"] === name) {
					this.loopJsonCache["operationalPolicies"][policy]["configurationsJson"] = newOpProperties;
				}
			}
	}

	updateOperationalPolicyPdpGroup(name, pdpGroup, pdpSubgroup) {
		for (var policy in this.loopJsonCache["operationalPolicies"]) {
			if (this.loopJsonCache["operationalPolicies"][policy]["name"] === name) {
				this.loopJsonCache["operationalPolicies"][policy]["pdpGroup"] = pdpGroup;
				this.loopJsonCache["operationalPolicies"][policy]["pdpSubgroup"] = pdpSubgroup;
			}
		}
	}

	getLoopName() {
		return this.loopJsonCache["name"];
	}

	getOperationalPolicyJsonSchema() {
		return this.loopJsonCache["operationalPolicies"]["0"]["jsonRepresentation"];
	}

	getOperationalPolicies() {
		return this.loopJsonCache["operationalPolicies"];
	}

	getOperationalPoliciesNoJsonSchema() {
		var operationalPolicies = JSON.parse(JSON.stringify(this.loopJsonCache["operationalPolicies"]));
		delete operationalPolicies[0]["jsonRepresentation"];
		return operationalPolicies;
	}

	getGlobalProperties() {
		return this.loopJsonCache["globalPropertiesJson"];
	}

	getDcaeDeploymentProperties() {
		return this.loopJsonCache["globalPropertiesJson"]["dcaeDeployParameters"];
	}

	getMicroServicePolicies() {
		return this.loopJsonCache["microServicePolicies"];
	}

	getOperationalPolicyForName(name) {
		var opProperties=this.getOperationalPolicies();
		for (var policy in opProperties) {
			if (opProperties[policy]["name"] === name) {
				return opProperties[policy];
			}
		}
		return null;
	}

	getOperationalPolicyPropertiesForName(name) {
		var opConfig = this.getOperationalPolicyForName(name);
		if (opConfig !== null) {
			return opConfig["configurationsJson"];
		}
		return null;
	}

	getOperationalPolicyJsonRepresentationForName(name) {
		var opConfig = this.getOperationalPolicyForName(name);
		if (opConfig !== null) {
			return opConfig["jsonRepresentation"];
		}
		return null;
	}

	getOperationalPolicySupportedPdpGroup(name) {
		var opConfig=this.getOperationalPolicyForName(name);
		if (opConfig !== null) {
		    if (opConfig["policyModel"]["policyPdpGroup"] !== undefined && opConfig["policyModel"]["policyPdpGroup"]["supportedPdpGroups"] !== undefined) {
			    return opConfig["policyModel"]["policyPdpGroup"]["supportedPdpGroups"];
			}
		}
		return [];
	}

	getOperationalPolicyPdpGroup(name) {
		var opConfig=this.getOperationalPolicyForName(name);
		if (opConfig !== null) {
			return opConfig["pdpGroup"];
		}
		return null;
	}

	getOperationalPolicyPdpSubgroup(name) {
		var opConfig=this.getOperationalPolicyForName(name);
		if (opConfig !== null) {
			return opConfig["pdpSubgroup"];
		}
		return null;
	}

	getMicroServiceSupportedPdpGroup(name) {
		var microService=this.getMicroServiceForName(name);
		if (microService !== null) {
		    if (microService["policyModel"]["policyPdpGroup"] !== undefined && microService["policyModel"]["policyPdpGroup"]["supportedPdpGroups"] !== undefined) {
			    return microService["policyModel"]["policyPdpGroup"]["supportedPdpGroups"];
			}
		}
		return [];
	}

	getMicroServicePdpGroup(name) {
		var microService=this.getMicroServiceForName(name);
		if (microService !== null) {
			return microService["pdpGroup"];
		}
		return null;
	}

	getMicroServicePdpSubgroup(name) {
		var microService=this.getMicroServiceForName(name);
		if (microService !== null) {
			return microService["pdpSubgroup"];
		}
		return null;
	}

	getMicroServiceForName(name) {
		var msProperties=this.getMicroServicePolicies();
		for (var policy in msProperties) {
			if (msProperties[policy]["name"] === name) {
				return msProperties[policy];
			}
		}
		return null;
	}

	getMicroServicePropertiesForName(name) {
		var msConfig = this.getMicroServiceForName(name);
		if (msConfig !== null) {
			return msConfig["configurationsJson"];
		}
		return null;
	}

	getMicroServiceJsonRepresentationForName(name) {
		var msConfig = this.getMicroServiceForName(name);
		if (msConfig !== null) {
			return msConfig["jsonRepresentation"];
		}
		return null;
	}

	getResourceDetailsVfProperty() {
		return this.loopJsonCache["modelService"]["resourceDetails"]["VF"];
	}

	getResourceDetailsVfModuleProperty() {
		return this.loopJsonCache["modelService"]["resourceDetails"]["VFModule"];
	}

	getLoopLogsArray() {
		return this.loopJsonCache.loopLogs;
	}

	getComputedState() {
		return this.loopJsonCache.lastComputedState;
	}

	getComponentStates() {
		return this.loopJsonCache.components;
	}

	getTemplateName() {
	    if (this.getLoopTemplate() !== undefined) {
        			return this.getLoopTemplate().name;
        }
	    return null;
	}

	getLoopTemplate() {
		return this.loopJsonCache["loopTemplate"];
	}

	isOpenLoopTemplate() {
		var loopTemplate = this.getLoopTemplate();
		if(loopTemplate != null && loopTemplate["allowedLoopType"] === "OPEN") {
			return true;
		}
		return false;
	}

	getAllLoopElementModels() {
	    var loopTemplate = this.getLoopTemplate();
	    var loopElementModels = [];
	    if(loopTemplate != null) {
	        for (var element of loopTemplate['loopElementModelsUsed']) {
	            loopElementModels.push(element['loopElementModel'])
	        }
	    }
	    return loopElementModels;
	}
}
