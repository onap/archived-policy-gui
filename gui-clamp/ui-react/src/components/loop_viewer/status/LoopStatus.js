/*-
* ============LICENSE_START=======================================================
* ONAP CLAMP
* ================================================================================
* Copyright (C) 2019 AT&T Intellectual Property. All rights
*                             reserved.
* ================================================================================
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
* http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* ============LICENSE_END============================================
* ===================================================================
*
*/
import React from 'react';
import Table from 'react-bootstrap/Table';
import styled from 'styled-components';
import LoopCache from '../../../api/LoopCache';

const LoopStatusViewDivStyled = styled.div`
	background-color: ${props => props.theme.loopViewerHeaderBackgroundColor};
	padding: 10px 10px;
	color: ${props => props.theme.loopViewerHeaderFontColor};
`

const TableStyled = styled(Table)`
    overflow: auto;
`

const TableRow = ({ statusRow }) => (
	<tr>
		<td>{statusRow.componentName}</td>
		<td>{statusRow.stateName}</td>
		<td>{statusRow.description}</td>
	</tr>

)

export default class LoopStatus extends React.Component {
	state = {
		loopCache: new LoopCache({})
	}

	constructor(props) {
		super(props);
		this.renderStatus = this.renderStatus.bind(this);
		this.state.loopCache = props.loopCache;
	}


	renderStatus() {
		if (this.state.loopCache.getComponentStates() != null) {
			return Object.keys(this.state.loopCache.getComponentStates()).map((key) => {
				console.debug("Adding status for: ",key);
				var res={}
				res[key]=this.state.loopCache.getComponentStates()[key];
				return (<TableRow statusRow={{'componentName':key,'stateName':this.state.loopCache.getComponentStates()[key].componentState.stateName,'description':this.state.loopCache.getComponentStates()[key].componentState.description}} />)
			})

		}
	}

	shouldComponentUpdate(nextProps, nextState) {
		return this.state.loopCache !== nextState.loopCache;
	}

	componentWillReceiveProps(newProps) {
		this.setState({
			loopCache: newProps.loopCache
		});
	}

	render() {
		return (
			<LoopStatusViewDivStyled>
				<label>Loop Status: {this.state.loopCache.getComputedState()}
				</label>

				<div >
					<TableStyled striped hover variant responsive>
						<thead>
							<tr>
								<th><span align="left">Component Name</span></th>
								<th><span align="left">Component State</span></th>
								<th><span align="right">Description</span></th>
							</tr>
						</thead>
						<tbody>
							{this.renderStatus()}
						</tbody>
					</TableStyled>
				</div>
			</LoopStatusViewDivStyled>
		);
	}
}

