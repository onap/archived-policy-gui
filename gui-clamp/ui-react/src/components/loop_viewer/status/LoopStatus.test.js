/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019 AT&T Intellectual Property. All rights
 *                             reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */
import React from 'react';
import { shallow } from 'enzyme';
import LoopStatus from './LoopStatus';
import LoopCache from '../../../api/LoopCache';

describe('Verify LoopStatus', () => {

  const loopCache = new LoopCache({
    "name": "LOOP_Jbv1z_v1_0_ResourceInstanceName1_tca",
    "lastComputedState": "DESIGN",
    "components": {
      "POLICY": {
        "componentState": {
          "stateName": "NOT_SENT",
          "description": "The policies defined have NOT yet been created on the policy engine"
        }
      },
      "DCAE": {
        "componentState": {
          "stateName": "BLUEPRINT_DEPLOYED",
          "description": "The DCAE blueprint has been found in the DCAE inventory but not yet instancianted for this loop"
        }
      }
    }
  });

  it('Test the render method', () => {
    const component = shallow(<LoopStatus loopCache={ loopCache }/>)

    expect(component).toMatchSnapshot();

    const loopCacheUpdated = new LoopCache({
      "name": "LOOP_Jbv1z_v1_0_ResourceInstanceName1_tca",
      "lastComputedState": "SUBMIT",
      "components": {
        "POLICY": {
          "componentState": {
            "stateName": "SENT",
            "description": "The policies defined have NOT yet been created on the policy engine"
          }
        },
        "DCAE": {
          "componentState": {
            "stateName": "BLUEPRINT_DEPLOYED",
            "description": "The DCAE blueprint has been found in the DCAE inventory but not yet instancianted for this loop"
          }
        }
      }
    });
    component.setProps({ loopCache: loopCacheUpdated });

    const forms = component.find('TableRow');
    expect(forms.get(0).props.statusRow.stateName).toEqual("SENT");
    expect(component.find('label').text()).toContain('SUBMIT');
  });
});
