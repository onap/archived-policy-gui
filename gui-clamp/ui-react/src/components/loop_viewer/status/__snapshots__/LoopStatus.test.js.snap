// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify LoopStatus Test the render method 1`] = `
<styled.div>
  <label>
    Loop Status: 
    DESIGN
  </label>
  <div>
    <Styled(Component)
      hover={true}
      responsive={true}
      striped={true}
      variant={true}
    >
      <thead>
        <tr>
          <th>
            <span
              align="left"
            >
              Component Name
            </span>
          </th>
          <th>
            <span
              align="left"
            >
              Component State
            </span>
          </th>
          <th>
            <span
              align="right"
            >
              Description
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <TableRow
          statusRow={
            Object {
              "componentName": "POLICY",
              "description": "The policies defined have NOT yet been created on the policy engine",
              "stateName": "NOT_SENT",
            }
          }
        />
        <TableRow
          statusRow={
            Object {
              "componentName": "DCAE",
              "description": "The DCAE blueprint has been found in the DCAE inventory but not yet instancianted for this loop",
              "stateName": "BLUEPRINT_DEPLOYED",
            }
          }
        />
      </tbody>
    </Styled(Component)>
  </div>
</styled.div>
`;
